jupyterhub/handlers/login.py

```diff
@@ -38,8 +40,11 @@ def _render(self, login_error=None, username=None):
     def get(self):
         self.statsd.incr('login.request')
         next_url = self.get_argument('next', '')
+        if next_url.startswith('%s://%s' % (self.request.protocol, self.request.host)):
+            # treat absolute URLs for our host as absolute paths:
+            next_url = urlparse(next_url).path
         if not next_url.startswith('/'):
```

Comment:
This was on purpose, but now that I think about it you're right. Done here as well.